src/katsdpcontroller/generator.py

```diff
@@ -208,6 +208,63 @@ def aggregate_compute(self) -> Tuple[Sensor.Status, int]:
         return (status, self._total)
 
 
+class SyncSensor(SimpleAggregateSensor[bool]):
+    """Aggregate which takes the logical AND of its children.
+
+    The status computes as ERROR if any of the child readings are False, and
+    FAILURE if they aren't all present.
```

Comment:
My gut feeling is that if an X-engine vanishes, it's safe to say that it's not synchronised. We are thus sure that the instrument is not synchronised (for which ERROR is appropriate) rather than being forced to say that we don't know.

(technically there are some failure modes where we lose comms to the X-engine but it's still running fine, but I think in such cases we should still say that it's not synchronised, just from a health-check point of view)